treadmill/infra/setup/master.py

```diff
@@ -0,0 +1,106 @@
+from treadmill.infra.vpc import VPC
+from treadmill.infra.instances import Instances
+from treadmill.infra.configuration import Configuration
+from pprint import pprint
+
+
+class MasterConfiguration(Configuration):
+    def __init__(self, domain, cell, app_root, freeipa_hostname, tm_release):
+        self.setup_scripts = [
+            {
+                'name': 'provision-base.sh',
+                'vars': {
+                    'DOMAIN': domain,
+                    'NAME': 'TreadmillMaster',
+                },
+            },
+            {'name': 'install-pid1.sh', 'vars': {}},
+            {'name': 'install-s6.sh', 'vars': {}},
+            {
+                'name': 'configure-master.sh',
+                'vars': {
+                    'DOMAIN': domain,
+                    'CELL': cell,
+                    'APPROOT': app_root,
+                    'FREEIPA_HOSTNAME': freeipa_hostname,
+                    'TREADMILL_RELEASE': tm_release,
+                },
+            },
+        ]
+        super(MasterConfiguration, self).__init__(self.setup_scripts)
+
+
+class Master:
+    def __init__(
+            self,
+            vpc_id=None,
+            domain='tw.treadmill',
+            app_root='/var/tmp'
+    ):
+        self.vpc = VPC(vpc_id, domain=domain)
+        self.domain = domain
+        self.app_root = app_root
+
+    def setup(
+            self,
+            tm_release='0.1.0',
+            freeipa_hostname='treadmillfreeipa'
```

Comment:
remove default, i think this is used elsewhere but in camelcase. create constant somewhere, but limit defaults to CLI.